results/cmd/api/main_test.go

```diff
@@ -437,3 +443,144 @@ func TestDeleteTaskRun(t *testing.T) {
+			Uid:       "543535",
+			Name:      "newtaskrun",
+			Namespace: "demo",
+		},
+	}
+	t5 := &pb.TaskRun{
+		ApiVersion: "v1alpha1",
+		Metadata: &pb.ObjectMeta{
+			Uid:       "543535",
+			Name:      "newtaskrun",
+			Namespace: "official",
+		},
+	}
+	taskruns := []*pb.TaskRun{t1, t2, t3, t4, t5}
+	for _, ts := range taskruns {
+		if _, err := srv.CreateTaskRunResult(ctx, &pb.CreateTaskRunRequest{
+			TaskRun: ts,
+		}); err != nil {
+			t.Fatalf("could not create taskrun: %v", err)
+		}
+	}
+	tt := []struct {
+		name         string
+		filter       string
+		expect       []*pb.TaskRun
+		expectStatus codes.Code
+	}{
+		{
+			name:         "test simple query",
+			filter:       `taskrun.api_version=="v1beta1"`,
+			expect:       []*pb.TaskRun{t1, t3, t4},
+			expectStatus: codes.OK,
+		},
+		{
+			name:         "test query with simple function",
+			filter:       `taskrun.metadata.name.endsWith("run")`,
+			expect:       []*pb.TaskRun{t1, t3, t4, t5},
+			expectStatus: codes.OK,
+		},
+		{
+			name:         "test empty filter",
+			filter:       "",
+			expect:       taskruns,
+			expectStatus: codes.OK,
+		},
+		{
+			name:         "test invalid field",
+			filter:       `task.name=="newtaskrun"`,
+			expectStatus: codes.InvalidArgument,
+		},
+		{
+			name:         "test invalid value type with no double quotes around value",
+			filter:       `taskrun.name==newtaskrun`,
+			expectStatus: codes.InvalidArgument,
+		},
+		{
+			name:         "test value not existing in the server record",
+			filter:       `taskrun.metadata.name=="notaskrun"`,
+			expect:       nil,
+			expectStatus: codes.OK,
+		},
+		{
+			name:         "test invalid field outside of our defined top level field",
+			filter:       `taskrun.unexistfield=="notaskrun"`,
+			expectStatus: codes.InvalidArgument,
+		},
+		{
+			name:         "test valid field but not boolean experssion",
+			filter:       `taskrun.api_version"`,
+			expectStatus: codes.InvalidArgument,
+		},
+		{
+			name:         "test random word input",
+			filter:       `tekton`,
+			expectStatus: codes.InvalidArgument,
+		},
+		{
+			name:         "test if field is case sensitive",
+			filter:       `taskrun.MetaData.name=="notaskrun"`,
+			expectStatus: codes.InvalidArgument,
+		},
+	}
+	// Test if we can find inserted taskruns
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			res, err := srv.ListTaskRunsResult(ctx, &pb.ListTaskRunsRequest{
+				Filter: tc.filter,
+			})
+			if res == nil && status.Code(err) != codes.InvalidArgument {
+				t.Fatalf("failed on test %v: %v", tc.name, err)
+			}
+			gotList := res.GetItems()
+			sort.SliceStable(gotList, func(i, j int) bool {
+				return gotList[i].String() < gotList[j].String()
+			})
+			sort.SliceStable(tc.expect, func(i, j int) bool {
+				return tc.expect[i].String() < tc.expect[j].String()
+			})
+			if diff := cmp.Diff(gotList, tc.expect, protocmp.Transform()); res != nil && status.Code(err) != codes.OK && diff != "" {
```

Comment:
If I'm understanding this, this will ignore checks in cases where the response is nil or there was no returned error. This effectively skips diffs for successful responses. I think we can just simplify this to check the responses, regardless of whether they are nil or not. Status checking can be handled in another statement (likely the one above).

```suggestion
			if diff := cmp.Diff(gotList, tc.expect, protocmp.Transform()); diff != "" {
```

Reply:
done